crates/metaslang/bindings/src/builder/loader/mod.rs

```diff
@@ -865,6 +852,14 @@ impl<'a, KT: KindTypes> Loader<'a, KT> {
         let node_handle = self.node_handle_for_graph_node(node_ref);
         let stack_graph_node = &self.stack_graph[node_handle];
 
+        let cursor = if let Some(source_node) = node.attributes.get(SOURCE_NODE_ATTR) {
+            let syntax_node_ref = source_node.as_syntax_node_ref()?;
+            let source_node = &self.graph[syntax_node_ref];
+            Some(source_node.clone())
```

Comment:
Q: is this `clone()` always needed?
If so, should we panic/error out if `stack_graph_node` is not a definition/reference below?

Reply:
nit suggestion for the future/not blocking:
WDYT of replacing the `is_definition()`/`is_reference()` APIs with a `get_kind() -> enum { Definition, Reference }`, to make sure call sites like this handle/check for all possible cases?